### PR TITLE
Remove previously deprecated API

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**:
Remove unsafe API that was deprecated last release.

**Motivation and Context**
Do the right thing

### CODE REVIEW

**Review by snnn — DISMISSED** [review]
Thank you!

File: include/onnxruntime/core/session/onnxruntime_cxx_api.h [hunk 1]
```diff
@@ -478,22 +443,14 @@ struct ModelMetadata : Base<OrtModelMetadata> {
...
 
-  /** \deprecated use GetCustomMetadataMapKeysAllocated()
-   * [[deprecated]]
-   * This interface produces multiple pointers that must be released
-   * by the specified allocator and is often leaked. Not exception safe.
+  /** \brief Returns a vector of copies of the custom metadata keys.
+   *
+   * \param allocator to allocate memory for the copy of the string returned
+   * \return a instance std::vector of smart pointers that would deallocate the buffers when out of scope.
+   *  The OrtAllocator instances must be valid at the point of memory release.
```
Comment (pranavsharma) [inline include/onnxruntime/core/session/onnxruntime_cxx_api.h:450]: nit: s/instances/instance

**Review by pranavsharma — APPROVED** [review]

File: objectivec/src/ort_session.mm [hunk 2]
```diff
@@ -186,7 +186,7 @@ - (BOOL)runWithInputs:(NSDictionary<NSString*, ORTValue*>*)inputs
     NSMutableArray<NSString*>* result = [NSMutableArray arrayWithCapacity:nameCount];
 
     for (size_t i = 0; i < nameCount; ++i) {
-      auto name = std::unique_ptr<char[], decltype(deleter)>{getName(i, allocator), deleter};
+      auto name = getName(i, allocator);
```
Comment (edgchen1) [inline objectivec/src/ort_session.mm:189]: can we also remove the `deleter` definition above?

**Review by edgchen1 — APPROVED** [review]

**Comment by fs-eire** [comment]
There are build failures for Web CI. Need to run `npm ci && npm run format` under /js/

File: js/web/lib/onnxjs/backends/webgl/inference-handler.ts [hunk 3]
```diff
@@ -260,9 +260,9 @@ export class WebGLInferenceHandler implements InferenceHandler {
   }
 
   private getTextureData(tensorId: Tensor.Id, isPacked = false): TextureData|undefined {
-    return this.session.isInitializer(tensorId) ?
-        this.session.getTextureData(tensorId, isPacked) :
-        isPacked ? this.packedTextureDataCache.get(tensorId) : this.unpackedTextureDataCache.get(tensorId);
+    return this.session.isInitializer(tensorId) ? this.session.getTextureData(tensorId, isPacked) :
+        isPacked                                ? this.packedTextureDataCache.get(tensorId) :
```
Comment (edgchen1) [inline js/web/lib/onnxjs/backends/webgl/inference-handler.ts:264]: nit: this formatting looks odd

**Review by edgchen1 — APPROVED** [review]

**Comment by snnn** [comment]
The samples in https://github.com/microsoft/onnxruntime-inference-examples also need to be updated.

**Pull request merged**